packages/app/src/AuthPage.tsx

```diff
@@ -9,27 +10,89 @@ import LandingHeader from './LandingHeader';
 import * as config from './config';
 import api from './api';
 
+type FormData = {
+  email: string;
+  password: string;
+};
+
 export default function AuthPage({ action }: { action: 'register' | 'login' }) {
+  const isRegister = action === 'register';
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+    setError,
+  } = useForm<FormData>({
+    reValidateMode: 'onSubmit',
+  });
   const router = useRouter();
   const { err, msg } = router.query;
 
   const { data: installation } = api.useInstallation();
 
   const verificationSent = msg === 'verify';
 
-  const title = `HyperDX - ${action === 'register' ? 'Sign up' : 'Login'}`;
+  const title = `HyperDX - ${isRegister ? 'Sign up' : 'Login'}`;
 
   useEffect(() => {
     // If an OSS user accidentally lands on /register after already creating a team
     // redirect them to login instead
-    if (
-      config.IS_OSS &&
-      installation?.isTeamExisting === true &&
-      action === 'register'
-    ) {
+    if (config.IS_OSS && installation?.isTeamExisting === true && isRegister) {
       router.push('/login');
     }
-  }, [installation, action, router]);
+  }, [installation, isRegister, router]);
+
+  const onSubmit: SubmitHandler<FormData> = async data => {
+    try {
+      const response = await fetch(`${API_SERVER_URL}/register/password`, {
```

Comment:
would be nice if we can wrap the api call using react query (check out `api.ts` file)